datalad_registry/tasks.py

```diff
@@ -33,7 +34,22 @@ def clone_dataset(url: str, ds_path: Path) -> Any:
         ds_repo.fetch(all_=True)
         ds_repo.call_git(["reset", "--hard", "refs/remotes/origin/HEAD"])
     else:
-        ds = dl.clone(url, ds_path_str)
+        max_incomplete_results_errs = 5
+        incomplete_results_err_count = 0
+
+        # Clone the dataset @ url into ds_path
+        while True:
+            try:
+                ds = dl.clone(url, ds_path_str)
+            except IncompleteResultsError:
+                lgr.warning(f"Incomplete results error in cloning {url}. Trying again.")
```

Comment:
`os.unlink(ds_path_str)` before retrying to ensure that follow up re-cloning is not somehow skipping since target exists but not fully kosher.